tests/std/tests/LWG3528_make_from_tuple_impl/test.compile.pass.cpp

```diff
@@ -0,0 +1,103 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//===----------------------------------------------------------------------===//
+
+#include <cstdint>
+#include <tuple>
+#include <type_traits>
+
+struct A {
+    int a;
+};
+struct B : public A {};
+
+struct C {
+    C(const B&) {}
+};
+
+enum class D {
+    ONE,
+    TWO,
+};
+
+template <class T, class Tuple>
+auto can_make_from_tuple(T&&, Tuple&& t) -> decltype(std::make_from_tuple<T>(t), uint8_t()) {
+    return 0;
+}
+template <class T, class Tuple>
+uint32_t can_make_from_tuple(...) {
+    return 0;
+}
+
+template <class T, class Tuple>
+inline constexpr bool has_make_from_tuple =
+    std::is_same_v<decltype(can_make_from_tuple<T, Tuple>(T{}, Tuple{})), uint8_t>;

```

Comment:
`T` and `Tuple` don't need to be initializable from `{}`. Perhaps we should include `<utility>` and use `std::declval` here.

Also, IIRC MSVC STL's test codes conventionally use SFINAE on partial specializations of variable templates.

Reply:
It may be desired to test "construction" of references.

Reply:
> T and Tuple don't need to be initializable from {}. Perhaps we should include <utility> and use std::declval here.

Done. and use `std::declval` now.
> Also, IIRC MSVC STL's test codes conventionally use SFINAE on partial specializations of variable templates.

I've added some additionally test to check variable template partial specialization.